### ISSUE TITLE
EF Core Migration Support

### ISSUE THREAD

**Issue by StevenRasmussen** [efcore/EFCore.SqlServer.HierarchyId#9]
When trying to create a migration, I get the following error:

> The current CSharpHelper cannot scaffold literals of type 'System.Data.SqlTypes.SqlBytes'. Configure your services to use one that can.

I see that from another issue, you mentioned that the `GenerateCodeLiteral` method needs to be overridden: [#1975](https://github.com/dotnet/efcore/issues/19785).

I inspected the source and it appears that this has not been overridden. See [here](https://github.com/efcore/EFCore.SqlServer.HierarchyId/blob/master/EFCore.SqlServer.HierarchyId/Storage/SqlServerHierarchyIdTypeMapping.cs).

I'd be happy to create a PR for this if that would help. Just let me know. Thanks!

**Comment by cutig3r** [comment]
I went ahead and got this working.  Attached is the project.  
[EFCore.SqlServer.HierarchyId.zip](https://github.com/efcore/EFCore.SqlServer.HierarchyId/files/4719269/EFCore.SqlServer.HierarchyId.zip)
Wasn't as simple as overriding the GenerateCodeLiteral.  Also, updated it to use the namespaces similar to [EFCore.SqlServer.NTS](https://github.com/dotnet/efcore/tree/master/src/EFCore.SqlServer.NTS)

What i didnt get to work was automatic usage of the designer, but in my test project (which houses the dbcontext), I added a class implementing IDesignTimeServices like this, and it picked it up:
```
using Microsoft.EntityFrameworkCore.Design;
using Microsoft.EntityFrameworkCore.SqlServer.Design.Internal;
using Microsoft.Extensions.DependencyInjection;

namespace HierarchyIdTests.Design
{
    public class HierarchyIdDesignTimeServices : IDesignTimeServices
    {
        private static readonly IDesignTimeServices hierarchyServices = new SqlServerHierarchyIdDesignTimeServices();

        /// <summary>
        ///     Configures design-time services. Use this method to override the default design-time services with your
        ///     own implementations.
        /// </summary>
        /// <param name="serviceCollection"> The design-time service collection. </param>
        public virtual void ConfigureDesignTimeServices(IServiceCollection serviceCollection)
        {
            hierarchyServices.ConfigureDesignTimeServices(serviceCollection);
        }
    }
}

```

**Comment by VivPat** [comment]
Having an issue with this too.

I applied the temporary fix above, but hoping that a final fix gets implemented into the master branch as soon as possible.

Nice work @cutig3r

**Comment by cutig3r** [comment]
@VivPat You may want to just go ahead and pull down my [fork](https://github.com/cutig3r/EFCore.SqlServer.HierarchyId/tree/hierarchyid-migrations), as this contains the latest & greatest.  I'm not sure if the one that I uploaded in that zip files has the proper code.  
PR #14 contains the latest code from my fork.

**Comment by bricelam** [comment]
Fixed by #14
